### ISSUE TITLE
do not warn owasp:api4:2019-string-restricted for strings with enum

### ISSUE THREAD

**Issue by DavidBiesack** [stoplightio/spectral-owasp-ruleset#28]
## Context

Apply the OWASP ruleset to an API which define properties with 

```yaml
type: string
enum:
  - val1
  - val2
  - val3
```

## Current Behavior

The ruleset issues a warning

`error  owasp:api4:2019-string-restricted        Schema of type string must specify a format or pattern.`

I find this to be overly strict. An `enum` on a string schema _does_ define an implicit pattern; i.e.
the above example is equivalent to 

```yaml
type: string
pattern: '^(val1|val2|val3)$'
```

but of course is much more palatable.

## Expected Behavior

to not emit an error or warning for strings with `enum` constraint.

## Possible Solution(s)

Not sure, but after a quick glance at the code, perhaps adding

```
              {
                required: ["enum"],
              },
```

near https://github.com/stoplightio/spectral-owasp-ruleset/blob/main/src/ruleset.ts#L544

(If I can carve off some time, I'll try a PR)

**Comment by DavidBiesack** [comment]
`const` should also be allowed to suppress the error
